doc/source/reference/c-api.ufunc.rst

```diff
@@ -93,12 +93,25 @@ Functions
         the corresponding 1-d loop function in the func array.
 
     :param types:
-        Must be of length (*nin* + *nout*) \* *ntypes*, and it
-        contains the data-types (built-in only) that the corresponding
-        function in the *func* array can deal with.
+
+        Length ``(nin + nout) * ntypes`` array of ``char`` encoding
+        the :ref:`PyArray_Descr.type_num` (built-in only) that the
+        corresponding function in the ``func`` array accepts. For
+        instance, for a comparison ufunc with three ``ntypes``, two
+        ``nin`` and one ``nout``, where the first function accepts
+        ``int32`` and the the second ``int64``, with both returning
+        booleans (``unsigned char``), ``types`` would be ``(char[])
```

Comment:
`(``unsigned char``)` is just wrong here. NPY_BOOL is its own type.

Reply:
Hmm, I got this from
```
grep npy_bool numpy/core/include/numpy/*.h
.
.
.
numpy/core/include/numpy/npy_common.h:typedef unsigned char npy_bool;
```

Reply:
`NPY_BOOL` is the enum value, which is distinct from `NPY_CHAR` - even if the same underlying C type is used to represent them, at the numpy level they are different.

Reply:
Maybe the description is a bit hinging on different things. One passes to the function definition those `NPY_` names, but inside the routine one would use `npy_bool`, `npy_int32,64`, etc.

Reply:
Maybe just change your description to use `npy_int32` and `npy_bool` - then you don't need to explain what those typedefs mean. If you single-backtick them, they _might_ link to their definitions.

Reply:
I think the `npy_*` are indeed linkable, though I cannot seem to build the docs - get a very helpful error message...
```
ValueError: The section Notes appears twice in the docstring of None in None.
```
Bit beyond this PR...

Reply:
@mhvk Hmm, last time I saw that error it was because masked array was modifying docstrings. I think that was fixed.